### PR TITLE
[READY]Automatic Miner Upgrade

### PR DESCRIPTION
**Pull request by SkeletalElite**
## About The Pull Request
Upgrade that fully automates miners, making human oversight only needed for defense/repairs.
Also I reused the sprites of the old compactor upgrade because it looked the part for the upgrade. Just renamed the files so it's less confusing
draft because needs #6210 merged first

## Why It's Good For The Game
Remote miners are often just not worth securing because you have to go so far out of your way. This upgrade makes more remote potentially valuable by allowing the miner to run without human oversight as long as it is not damaged by xenos. Also is a good replacement for the compactor upgrade in terms of function (improves logistics) since the compactor will become obsolete if the prerequisite PR is merged.

## Changelog
:cl:
add: New miner upgrade that automates the miner.
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
If you want to test and draft something that needs another PR to be merged, you can make a new branch out of your main miner branch for exemple.

That way, you can do all the changes you need, merge from the miner branch when changes are made to it, etc...

When this PR will be ready to review an the other PR merge, you can just rebase this branch onto the master, and the commits will be clean

**Comment by MLGTASTICa** [comment]
Make it a req-only kit perhaps ? This takes out the risk of having to carry the crates back and the miners of ever being undefended . If marines want to acquire points they should put some money or manpower into it..

**Comment by SkeletalElite** [comment]
> Make it a req-only kit perhaps ? This takes out the risk of having to carry the crates back and the miners of ever being undefended . If marines want to acquire points they should put some money or manpower into it..

The prerequisite PR for this removes crates entirely, miners just generate points

Also, miners being undefended is a risk of it's own, if the miner isn't defended than all it takes is 1 runner or hunter to run in and slash it. The upgrade also isn't completely free, requiring 18 metal and an autolathe to create. 
The cost of producing them isn't very high, but there's still a time investment and they can only be made on the ship. 
You also can't use an overclock if you're using this upgrade.

**Comment by MLGTASTICa** [comment]
> > Make it a req-only kit perhaps ? This takes out the risk of having to carry the crates back and the miners of ever being undefended . If marines want to acquire points they should put some money or manpower into it..
> 
> The prerequisite PR for this removes crates entirely, miners just generate points
> 
> Also, miners being undefended is a risk of it's own, if the miner isn't defended than all it takes is 1 runner or hunter to run in and slash it. The upgrade also isn't completely free, requiring 18 metal and an autolathe to create.
> The cost of producing them isn't very high, but there's still a time investment and they can only be made on the ship.
> You also can't use an overclock if you're using this upgrade.

Then fix the issue at cause , runners being able to instantly slash away a undefended miner without effort , perhaps implement an alien act that takes some time to saboutage and exposes them to any marine that is there to defend it . This  way xenos that rush them alone can be punished. Miners shouldnt just magically teleport points to req without any effort on grounside or any investment from req. Req is balanced all around this and if the change goes  past , a rebalance of the credits per box would be necesarry , perhaps 5 credits instead of 20.

**Comment by SkeletalElite** [comment]
> Then fix the issue at cause , runners being able to instantly slash away a undefended miner without effort , perhaps implement an alien act that takes some time to saboutage and exposes them to any marine that is there to defend it . This way xenos that rush them alone can be punished. Miners shouldnt just magically teleport points to req without any effort on grounside or any investment from req. Req is balanced all around this and if the change goes past , a rebalance of the credits per box would be necesarry , perhaps 5 credits instead of 20.

If it ends up being necessary nerfing the value of minerals is literally a 2 line change (1 for plat 1 for phoron) but like I said in the other PR , we want to push gameplay about capturing multiple miners, not sitting on 1, that's just not viable with the amount of effort required to run a miner, and is also the reason only miners near the tadpole or FOB are captured, miners that aren't close just are not viable. This should make miners that aren't immediately near the FOB or tadpole usable.

**Comment by BraveMole** [comment]
> Then fix the issue at cause , runners being able to instantly slash away a undefended miner without effort , perhaps implement an alien act that takes some time to saboutage and exposes them to any marine that is there to defend it . This way xenos that rush them alone can be punished. Miners shouldnt just magically teleport points to req without any effort on grounside or any investment from req. Req is balanced all around this and if the change goes past , a rebalance of the credits per box would be necesarry , perhaps 5 credits instead of 20.

The issue at cause is babysitting miners is boring. Like skel said, if this update means marines have a way too much points, just make miners give less points.

Let's try to think about fun, then balance

**Comment by SkeletalElite** [comment]
If a kind maintainer could remove the DNM from this that would be great

**Comment by SplinterGP** [comment]
This is actually good and i see it balanced bcos overclock also exists and those generate more points but you gotta come on them every 4 minutes

File: code/game/objects/items/misc.dm [hunk 1]
```diff
@@ -138,3 +138,9 @@
 	icon_state = "mining_drill_overclockeddisplay"
 	uptype = "high-efficiency drill"
 
+
+obj/item/minerupgrade/automatic
```
Comment (TiviPlus) [inline code/game/objects/items/misc.dm:142]: Missing /

File: code/game/objects/machinery/miner.dm [hunk 2]
```diff
@@ -216,7 +224,9 @@
 	if(miner_status != MINER_RUNNING)
 		to_chat(user, "<span class='warning'>[src] is damaged!</span>")
 		return
-
+	if(miner_upgrade_type == "mining computer")
```
Comment (TiviPlus) [inline code/game/objects/machinery/miner.dm:227]: Use defines for this

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by SkeletalElite** [comment]
Now I remember why I wasn't using the defines in the first place, they were always returning true even when the upgrade wasn't installed

File: code/game/objects/machinery/miner.dm (hunk 2 above)
Comment (SkeletalElite) [reply code/game/objects/machinery/miner.dm]: The define is always returning true here what do

File: code/game/objects/machinery/miner.dm [hunk 3]
```diff
@@ -77,6 +77,14 @@
...
 		if(MINER_OVERCLOCKED)
 			required_ticks = 60
+		if(MINER_AUTOMATED)
+			if(stored_mineral)
+				SSpoints.supply_points += mineral_value * stored_mineral
+				do_sparks(5, TRUE, src)
+				playsound(loc,'sound/effects/phasein.ogg', 50, FALSE)
+				say("Ore shipment has been sold for [mineral_value * stored_mineral] points.")
+				stored_mineral = 0
+				start_processing()
```
Comment (TiviPlus) [inline code/game/objects/machinery/miner.dm:87]: This can cause machines to start processing twice

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
also if this is too good you can just reduce the speed of a miner

**Review by TiviPlus — APPROVED** [review]

File: code/game/objects/items/misc.dm [hunk 4]
```diff
@@ -156,3 +156,9 @@
 /obj/item/dropship_points_voucher/examine(mob/user)
 	. = ..()
 	to_chat(user, "It contains [extra_points] points.")
+
+/obj/item/minerupgrade/automatic
+	name = "mining computer"
+	desc = "A small computer that can automate mining wells, reducing the need for oversight"
```
Comment (XSlayer300) [inline code/game/objects/items/misc.dm:162]:
```suggestion
	desc = "A small computer that can automate mining wells, reducing the need for oversight."
```

File: code/game/objects/machinery/miner.dm [hunk 5]
```diff
@@ -216,7 +224,9 @@
 	if(miner_status != MINER_RUNNING)
 		to_chat(user, "<span class='warning'>[src] is damaged!</span>")
 		return
-
+	if(miner_upgrade_type == MINER_AUTOMATED)
+		to_chat(user, "<span class='warning'>[src] is automated!")
```
Comment (XSlayer300) [inline code/game/objects/machinery/miner.dm:228]:
```suggestion
		to_chat(user, "<span class='warning'>[src] is automated!"</span>)
```

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**